docs/source/scilla-in-depth.rst

```diff
@@ -802,6 +830,233 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
+the contents of that address matches the address type (in this
+case that the address is in use either by a user or by a
+contract), then deployment continues, and ``init_owner`` can be
+treated as a ``ByStr20 with end`` throughout the contract.
+
+Similarly, a transition might specify a parameter
+``token_contract`` as follows:
+
+.. code-block:: ocaml
+                
+      transition Transfer (
+          token_contract : ByStr20 with contract
+                                          field balances : Map ByStr20 Uint128
+                                   end
+          )
+
+When the transition is invoked, the byte string supplied as the
+``token_contract`` parameter is looked up as an address on the
+blockchain, and if the contents of that address matches the address
+type (in this case that the address contains a contract with a
+field ``balances`` of a type that is assignable to ``Map ByStr20
+Uint128``), then the transition parameter is initialised
+successfully, and ``token_contract`` can be treated as a ``ByStr20
+with contract field balances : Map ByStr20 Uint128 end`` throughout
+this transition invocation.
+
+In either case, if the contents of the address does not match the
+specified type, then the dynamic typecheck is unsuccessful, causing
+deployment (for failed immutable parameters) or transition
+invocation (for transition parameters) to fail. A failed dynamic
+typecheck is considered a run-time error, causing the current
+transaction to abort. (For the purposes of dynamic typechecks of
+immutable fields the deployment of a contract is considered a
+transaction).
+
+.. note::
+
+   It is not possible to specify a ``ByStr20`` literal and have it
+   interpreted as an address. In other words, the following code
+   snippet will result in a static type error:
+
+   .. code-block:: ocaml
+                
+                   let x : ByStr20 with end = 0x1234567890123456789012345678901234567890
+   
+   The only way for a byte string to be validated against an address
+   type is to pass it as the value of an immutable field or as a
+   transition parameter, of the appropriate type.
+
+   
+Remote fetches
+--------------
+  
+To perform a remote fetch ``x <- & c.f``, the type of ``c`` must be
+some address type declaring the field ``f``. For instance, if ``c``
+has the type ``ByStr20 with contract field paused : Bool end``, then
+the value of the field ``paused`` at address ``c`` can be fetched
+using the statement ``x <- & c.paused``, whereas it is not possible to
+fetch the value of an undeclared field (e.g., ``admin``) of ``c``,
+even if the contract at address ``c`` does actually contain a field
+``admin``. To be able to fetch the value of the ``admin`` field, the
+type of ``c`` must contain the ``admin`` field as well, e.g, ``ByStr20
+with contract field paused : Bool, field admin : ByStr20 end``
+
+Remote fetches of maps fields can be performed using in-place
```

Comment:
```suggestion
Remote fetches of maps' fields can be performed using in-place
```

Reply:
Should be "map fields". Otherwise it's "fields inside a map", which makes no sense.